### PR TITLE
Update Go version to 1.12

### PR DESCRIPTION
**Pull request by gunnaraasen**
This PR updates the Go runtime used to build InfluxDB to Go 1.12.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by e-dard** [comment]
Wondering what broke in CI?

**Comment by gunnaraasen** [comment]
This is failing because the build cache is [required in Go 1.12](https://golang.org/doc/go1.12#gocache) and go does not have permissions to create a `~/.cache` directory in the home directory. Seems like other people have set `GOCACHE=/tmp`. I'm not very familiar with our Jenkins pipeline. Where is the right place to set `GOCACHE`?

**Comment by gunnaraasen** [comment]
@gshif resolved the CI error. @e-dard, can you take another look?

File: test.sh [hunk 1]
```diff
@@ -124,8 +124,8 @@ case $ENVIRONMENT_INDEX in
         rc=$?
         ;;
     4)
-        # go1.11
-        run_test_docker Dockerfile_build_ubuntu64_go1.11 test_64bit --test --junit-report
+        # go1.12
+        run_test_docker Dockerfile_build_ubuntu64_go1.12 test_64bit --test --junit-report
```
Comment (mark-rushakoff) [inline test.sh]: I'm not sure if this is needed now, or if we should be using 1.13beta for this. The previous stages should be testing with 1.12 already so this seems like duplicate work.

Comment (gunnaraasen) [reply test.sh]: This line should be updated to 1.13beta along with the [release scripts](https://github.com/influxdata/influxdb/blob/1.8/releng/_go_versions.sh#L5). Go 1.13beta won't be out until August 1st though.

Comment (mark-rushakoff) [reply test.sh]: 1.13beta1 has already been released: https://groups.google.com/forum/?pli=1#!topic/golang-dev/Qn7Gc9-y-nw

1.13beta images appear to be available on Docker hub.

**Comment by gunnaraasen** [comment]
@mark-rushakoff I just pushed up an Appveyor config update.

**Pull request merged**